tests/upgrade/test-rc9-pass2-backfill.sh

```
@@ -0,0 +1,269 @@
+# When WORKDIR_OLD is available and the project file byte-matches the baseline,
+# the migration skips it (clean canonical fast-path); when not, it falls through
+# to the slug check. We omit WORKDIR_OLD here to exercise the slug-check path.
+#
+# Each post-migration agent file is validated with lint-agent-contracts.sh
+# --canonical-only to confirm it satisfies the rc9 contract schema.
+
+set -u
+
+repo_root="$(cd "$(dirname "$0")/../.." && pwd)"
+migration="$repo_root/migrations/v1.0.0-rc9.sh"
+lint_script="$repo_root/scripts/lint-agent-contracts.sh"
```

Shellcheck (0.11.0)

[warning] lint_script appears unused. Verify use (or export if used externally).

(SC2034)